users/serializers.py

```diff
@@ -0,0 +1,77 @@
+import django.contrib.auth.password_validation as password_validation
+from django.contrib.auth.models import User
+from rest_framework import serializers
+from rest_framework.authtoken.models import Token
+
+
+class UserDetailSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username',)
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username', 'password')
+
+    def save(self):
+        user = User.objects.create_user(self.data["username"])
+        user.set_password(self.data["username"])
+        if "email" in self.data:
+            user.email = self.data["email"]
+
+        user.save()
+
+
+class LoginSerializer(serializers.Serializer):
```

Comment:
This serializer visits the database twice to get the user. I believe you can optimize this.